pkg/generate/config/field.go

```diff
@@ -105,6 +105,17 @@ type SourceFieldConfig struct {
 	Path string `json:"path"`
 }
 
+// CompareFieldConfig informs the code generator how to compare two values of a
+// field
+type CompareFieldConfig struct {
+	// IsIgnored indicates the field should be ignored when comparing a
+	// resource
+	IsIgnored bool `json:"is_ignored"`
+	// NilEqualsEmptyString indicates a nil string pointer and empty string
+	// should be considered equal for the purposes of comparison
+	NilEqualsEmptyString bool `json:"nil_equals_empty_string"`
```

Comment:
I'm wondering if we should rename this field to `NilEqualsZeroValue`? i'm thinking about cases where we want to equate `0` and `nil`, `0.0` and `nil`, `false` and `nil` ... etc

Reply:
awesome point.